### ISSUE TITLE
Can't type 'enter' in text editors inside dialogs

### ISSUE THREAD

**Issue by yuvipanda** [jupyterlab/jupyterlab#5802]
If a CodeEditor (in my case, a JSONEditor) is placed inside a Dialog, you can not create a newline in the code editor with the enter key. This seems to be because the Dialog unconditionally catches the 'Enter' (https://github.com/jupyterlab/jupyterlab/pull/1652) and doesn't let the event go to the editor itself.

**Comment by jasongrout** [comment]
In particular, the event handlers are registered for the capture phase, so the events never get down to the elements.

**Comment by alfonsoar** [comment]
Does anyone have any recommendations on a workaround? 

I would ideally like to remove the event handler but since I don't have to the initial named function (`this` in this case) passed to `addEventListener()` my attempts to remove the event listener have failed. 

This makes it hard to have forms inside the Dialog.

**Comment by jasongrout** [comment]
How about a PR that makes the keydown handler not be on the capture phase?

**Comment by alfonsoar** [comment]
Yeah I can work on that, can we backport it to version `1.2.x` ? I am on version `1.2.8` currently and have some blockers to update.

**Comment by alfonsoar** [comment]
@jasongrout - is it possible?
